Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -488,7 +489,14 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
             } else {
                 cell.favicon.image = UIImage(named: "defaultFavicon")
             }
-            cell.background.image = tab.screenshot
+
+            if let img = tab.screenshot {
+                cell.background.image = img
+            } else if let thumbs = profile.thumbnails as? SDWebThumbnails {
+                let key = SDWebThumbnails.getKey(tab.url!)
+                cell.background.moz_getImageFromCache(key, cache: thumbs.cache, completed: { img, err, type, key in
+                })
+            }
```

Comment:
I forgot those are half size. I'd still rather we just stored them normal size once than this. i.e. one place. one api. etc.